packager/file/file.h

```diff
@@ -25,7 +25,7 @@ extern const char* kHttpFilePrefix;
 const int64_t kWholeFile = -1;
 
 /// Define an abstract file interface.
-class SHAKA_EXPORT File {
+class SHAKA_EXPORT File : private DisallowCopyAndAssign {
```

Comment:
I disagree with your reading of the style guide.  Specifically, these two parts seem to support using this pattern.

> Every class's public interface must make clear which copy and move operations the class supports.

and

> These declarations/deletions can be omitted only if they are obvious: [...] If a base class clearly isn't copyable or movable, derived classes naturally won't be either.

I would argue that this makes it both clear and obvious what is allowed, and it's much easier to see and repeat IMO than explicit deletions.  I don't think it's really much different than what we did before with the Chrome macro, and to me, it's even more visible because it's right at the top.

Also, this is one of the patterns recommended by ABSL: https://abseil.io/tips/143

Reply:
I don't think it is allowed inside Google code to define a generic top-level base class for DisallowCopyAndAssign, otherwise we would have seen it in Google3 or in ABSL. It adds an unnecessary dependency without benefits (over explicit definition in the "actual" top level base class). Google C++ style requires "non-copyable/movable class should explicitly delete the copy operations", although it allows them to be omitted for a few special cases (including the case where the base class "clearly" isn't copyable / moveable).

Since Shaka Packager is an independent project, I don't really have a strong opinion on the style compliance.

> this is one of the patterns recommended by ABSL: https://abseil.io/tips/143

It is an "old" style from boost, not really the pattern recommended by ABSL.

Reply:
Okay, then I'll write a macro to replace the one from Chromium and revert to that style.

Reply:
Done